### PR TITLE
[1.20] oci: finish reverting exec changes

### PR DESCRIPTION
**Pull request by haircommander**
as well as fix an issue with terminals in execs

Signed-off-by: Peter Hunt <pehunt@redhat.com>
/kind bug

#### Does this PR introduce a user-facing change?

```release-note
fix an issue when a container with a terminal has an exec probe

```

### CODE REVIEW

**Comment by haircommander** [comment]
/retest
(quay flake)

**Comment by haircommander** [comment]
related https://bugzilla.redhat.com/show_bug.cgi?id=1985539

File: internal/oci/runtime_oci.go [hunk 1]
```diff
@@ -262,7 +262,7 @@ func (r *runtimeOCI) ExecContainer(ctx context.Context, c *Container, cmd []stri
 	}
 	defer os.RemoveAll(processFile)
 
-	execCmd := r.constructExecCommand(ctx, c, processFile, "")
+	execCmd := r.constructExecCommand(ctx, c, processFile, "", tty)
```
Comment (mrunalp) [inline internal/oci/runtime_oci.go:265]: We are missing this in release-1.21.

Comment (haircommander) [reply internal/oci/runtime_oci.go]: oh yes this is not actually needed anymore

Comment (haircommander) [reply internal/oci/runtime_oci.go]: updated

**Review by mrunalp — APPROVED** [review]
/lgtm

**Pull request merged**